### ISSUE TITLE
When using an anonymous struct in slint, the generated rust type doesn't preserve the order of the members

### ISSUE THREAD

**Issue by Altyrost** [slint-ui/slint#6756]
# System Infos
OS: Linux Mint 22 
arch: x64
slint version: 1.8.0
slint build version: 1.8.0

# Actual behavior

  When using an anonymous struct in slint, the generated type doesn't preserve the order of the members. This causes problems when the user tries to expose a tuple through a slint::Model. 

# Expected behavior

  When using an anonymous struct in slint, the generated types should preserve the order of the members. 

# How to reproduce
 
 In main.rs
```rust
// Prevent console window in addition to Slint window in Windows release builds when, e.g., starting the app via file manager. Ignored on other platforms.
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]

use std::error::Error;
use std::rc::Rc;
use slint::ModelTracker;

slint::include_modules!();

struct Item {
    uuid: usize,
    description: String,
}

struct TodoList {
    items: Vec<Item>
}

impl TodoList {
    fn new() -> Self {
        TodoList {
            items: Vec::new()
        }
    }

    fn add(&mut self, description: String) {
        self.items.push(Item {
            uuid: self.items.len(),
            description
        })
    }
}

impl slint::Model for TodoList {
    type Data = (i32, slint::SharedString);

    fn row_count(&self) -> usize {
        self.items.len()
    }

    fn row_data(&self, row: usize) -> Option<Self::Data> {
        let item = self.items.get(row)?;
        Some((
            item.uuid as i32,
            slint::SharedString::from(item.description.clone())
        ))
    }

    fn model_tracker(&self) -> &dyn ModelTracker {
        &()
    }
}


fn main() -> Result<(), Box<dyn Error>> {
    let mut todo = TodoList::new();
    for i in 0..10 {
        todo.add(std::format!("test {}", i));
    }

    let todo_rc = Rc::new(todo);
    let ui = AppWindow::new()?;
    ui.set_todolist(slint::ModelRc::from(todo_rc.clone()));
    ui.run()?;
    Ok(())
}
```

in app-window.slint
```slint
import { HorizontalBox, ListView } from "std-widgets.slint";

export component AppWindow inherits Window {
    in property <[{uuid: int, description: string}]> todolist;

    ListView {
        for item in todolist : HorizontalBox {
            Text {
                text: item.uuid;
            }
            Text {
                text: item.description;
            }
        }
    }
}
```

The error: 
```
error[E0308]: mismatched types
    --> src/main.rs:63:21
     |
63   |     ui.set_todolist(slint::ModelRc::from(todo_rc.clone()));
     |        ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `ModelRc<(SharedString, i32)>`, found `ModelRc<(i32, SharedString)>`
     |        |
     |        arguments to this method are incorrect
     |
     = note: expected struct `ModelRc<(slint::SharedString, i32)>`
                found struct `ModelRc<(i32, slint::SharedString)>`
```

**Comment by ogoffart** [comment]
In the Slint type system, there is no difference between: `{uuid: int, description: string}` and `{description: string, uuid: int}` and in the generated code, Slint create a tuple with the field ordered by their name.  (so description before uuid).

I recommend to use named struct when a property is exposed to the native language:

```slint
 export struct TodoItem{uuid: int, description: string}
 export component AppWindow inherits Window {
    in property <[TodoItem]> todolist;
 }
``` 

Changing the anonymous struct type at this point would be a breaking change. And would also make code generation more complicated as there would be more location where we have to convert.

**Comment by ogoffart** [comment]
I'm going to close this issue as this would be a breaking change.
But thanks anyway for filing it.

**Comment by FaintentDev** [comment]
i couldnt agree about this more, and its giving me a hard time when using simple anonymous structs within actual structs cause i want to see the tuple in the same order as i have the fields in rust. sad to see it wont change at this point

**Comment by FaintentDev** [comment]
you guys could just throw this in as a breaking fix in 1.14, just a suggestion
